http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -565,12 +540,15 @@ object LeonardoJsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
             fields.getOrElse("clusterName", JsNull).convertTo[ClusterName],
             fields.getOrElse("googleProject", JsNull).convertTo[GoogleProject],
             fields.getOrElse("serviceAccountInfo", JsNull).convertTo[ServiceAccountInfo],
-            DataprocInfo(
-              fields.getOrElse("googleId", JsNull).convertTo[Option[UUID]],
-              fields.getOrElse("operationName", JsNull).convertTo[Option[OperationName]],
-              fields.getOrElse("stagingBucket", JsNull).convertTo[Option[GcsBucketName]],
-              fields.getOrElse("hostIp", JsNull).convertTo[Option[IP]]
-            ),
+            (fields.get("googleId"), fields.get("operationName"), fields.get("stagingBucket")).mapN {
+              (googleId, operationName, stagingBucket) =>
+                DataprocInfo(
+                  googleId.convertTo[UUID],
+                  operationName.convertTo[OperationName],
+                  stagingBucket.convertTo[GcsBucketName],
+                  fields.getOrElse("hostIp", JsNull).convertTo[Option[IP]]
+                )
```

Comment:
semantic is a bit different between this and the original code...This code will return `None` if one of `googleId`, `operationName`, `stagingBucket` is `None`, vs the original logical seems to indicate these fields are not `None` or `Some` at the same time

Reply:
Yeah this is a semantic change.. `DataprocInfo` is optional now, so I was saying to only construct it if `googleId`, `operationName`, and `stagingBucket` are present in the input json (which are required fields in `DataprocInfo`). I think this is valid based on the data model.. and actually, this json reader is never actually used since we never serialize input JSON to `Cluster` objects.

Reply:
cool cool...That's way better than what it was before...Just wanted to call out the semantic change explicitly :)